### DISCUSSION TITLE
[General] `eth_getTransactionReceipt` return wrong contractAddress

### DISCUSSION THREAD

**Discussion by zouguangxian** [zkSync-Community-Hub/zksync-developers#367] [General]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement.

-

### What is the specific issue or error you're encountering?

   I deployed multiple contracts on zksync sepolia, and I found that sometimes the `contractAddress` returned by `eth_getTransactionReceipt` is wrong.

Here are two examples of the same contracts.

- This transaction `0xa2c41fc7307720cd5fed3696abe9db1a0d428d275b9762632450c32ad5b9010d` returns the wrong `contractAddress` 0x6A5c0c4361e22b91F65052E29D5660613A28c125, the correct contractAddress should be 0xd0789a7e9Ae0015BBF5c82f365eE860261e70e5D.

curl --location 'https://sepolia.era.zksync.dev' \
--header 'Content-Type: application/json' \
--data '{
  "jsonrpc":"2.0",
  "method":"eth_getTransactionReceipt",
  "params":[
    "0xa2c41fc7307720cd5fed3696abe9db1a0d428d275b9762632450c32ad5b9010d"
  ],
  "id":1
}'

- However, This transaction `0xf289ef298e14cccdc7035e97ae65460ee17b01a3730a04adecedb8c0d790af2c` returns the correct contractAddress 0x84565bCbA0Eed893a2b17c69Cca44a3eAFAC01E4.
curl --location 'https://sepolia.era.zksync.dev' \
--header 'Content-Type: application/json' \
--data '{
  "jsonrpc":"2.0",
  "method":"eth_getTransactionReceipt",
  "params":[
    "0xf289ef298e14cccdc7035e97ae65460ee17b01a3730a04adecedb8c0d790af2c"
  ],
  "id":1
}'

Replace the https://sepolia.era.zksync.dev with
https://zksync-era-sepolia.blockpi.network/v1/rpc/public to generate the same result.

You can find that the bytecode of contracts at 0x84565bCbA0Eed893a2b17c69Cca44a3eAFAC01E4 and 0xd0789a7e9Ae0015BBF5c82f365eE860261e70e5D are the same.

### Can you share the error messages or logs you're receiving, if any?

-

### Have you made any recent changes to the contract before encountering this issue?

-

### Are there any external libraries or contracts that your contract interacts with?

-

### Can you provide the relevant portions of your contract code where the issue is occurring?

-

### Have you tried to isolate the problem, and if so, what were the results?

-

### What steps have you already taken to try to resolve the issue?

-

**Comment by quanteauphil** [comment]
The issue seems to be related to inconsistencies in the contractAddress returned by eth_getTransactionReceipt on zkSync sepolia testnet. Despite deploying the same contract bytecode, different transactions return different contract addresses. Further investigation is needed to understand the underlying cause of this inconsistency and potential fixes.

**Reply by zouguangxian** [reply]
@quanteauphil Thanks. It looks like the second item of logs contains the correct deployAddress, and the wrong contractAddress is for another existing contract.

**Comment by tiovladelfabs** [comment]
It appears that there may be inconsistencies in the contract deployment process on the zksync sepolia testnet, leading to incorrect contract addresses returned by eth_getTransactionReceipt. This could be due to various factors such as network issues or bugs in the deployment process. It's recommended to reach out to the zksync team for further assistance and investigation.

**Comment by webgolfrelenromero** [comment]
This issue seems to be related to the inconsistency in returning the contract address by eth_getTransactionReceipt on zkSync sepolia. It's essential to investigate and address this inconsistency to ensure accurate contract deployment tracking.

**Comment by RomanBrodetski** [comment]
Hello, 
I confirm that there is a bug where instead of returning the address of the **last** contract that was deployed by transaction, in some cases we return an arbitrary one. 

We will resolve this and follow up.

**Reply by zouguangxian** [reply]
Thank you very much. I’m glad to hear that root cause of this issue has been identified.

**Reply by RomanBrodetski** [reply]
just quick note - while we are fixing this, there is a workaround: you can find the deployed contract address in the events that the transaction emitted. 
Not sure it's worth for you since it will be fixed soon, still wanted to let you know

**Reply by zouguangxian** [reply]
Thanks for your help. I will continue based on the final solution.

**Comment by AnastasiiaVashchuk** [comment]
Hi there,
Happy to inform that the issue has been fixed.

**Reply by zouguangxian** [reply]
Awesome, thank you very much.

**Comment by zouguangxian** [comment]
> Hi there,
> 
> Happy to inform that the issue has been fixed.

We tested it on the Sepolia multiple times, and the failure disappeared. Thanks for your great work.
